docs/m4t/README.md

```diff
@@ -39,8 +39,199 @@ We provide the extensive evaluation results of seamlessM4T-Large and SeamlessM4T
+
+```python
+# Save the translated audio output:
+import torchaudio
+torchaudio.save(
+    <path_to_save_audio>,
+    speech_output.audio_wavs[0][0].cpu(),
+    sample_rate=speech_output.sample_rate,
+)
+```
+**S2TT, T2TT and ASR (text output):**
+
+```python
+# S2TT
+text_output, _ = translator.predict(
+    input=<path_to_input_audio>, 
+    task_str="S2TT", 
+    tgt_lang=<tgt_lang>, 
+    text_generation_opts=text_generation_opts, 
+    unit_generation_opts=None
+)
+
+# ASR
+# This is equivalent to S2TT with `<tgt_lang>=<src_lang>`.
+    text_output, _ = translator.predict(
+    input=<path_to_input_audio>, 
+    task_str="ASR", 
+    tgt_lang=<src_lang>, 
+    text_generation_opts=text_generation_opts, 
+    unit_generation_opts=None
+)
+
+# T2TT
+text_output, _ = translator.predict(
+    input=<input_text>, 
+    task_str="T2TT", 
+    tgt_lang=<tgt_lang>, 
+    src_lang=<src_lang>, 
+    text_generation_opts=text_generation_opts, 
+    unit_generation_opts=None
+)
+
+```
+Note that `<src_lang>` must be specified for T2TT
+
+To reproduce the seamless papers results ([v1](https://arxiv.org/abs/2308.11596) or [v2](https://arxiv.org/abs/2312.05187)), or to evaluate using the same metrics over your own test sets, please check out the [Evaluation README here](../../src/seamless_communication/cli/m4t/evaluate/README.md).
+
+## Inference with 🤗 `Transformers`
+
+SeamlessM4T is available in the 🤗 Transformers library, requiring minimal dependencies. Steps to get started:
+
+1. First install the 🤗 [Transformers library](https://github.com/huggingface/transformers) from main and [sentencepiece](https://github.com/google/sentencepiece):
+
+```
+pip install git+https://github.com/huggingface/transformers.git sentencepiece
+```
+
+2. Run the following Python code to generate speech samples. Here the target language is Russian:
+
+```py
+import torchaudio
+from transformers import AutoProcessor, SeamlessM4Tv2Model
+
+processor = AutoProcessor.from_pretrained("facebook/seamless-m4t-v2-large")
+model = SeamlessM4Tv2Model.from_pretrained("facebook/seamless-m4t-v2-large")
+
+# from text
+text_inputs = processor(text="Hello, my dog is cute", src_lang="eng", return_tensors="pt")
+audio_array_from_text = model.generate(**text_inputs, tgt_lang="rus")[0].cpu().squeeze()
+
+# from audio
+audio, orig_freq = torchaudio.load("https://www2.cs.uic.edu/~i101/SoundFiles/preamble10.wav")
+audio = torchaudio.functional.resample(audio, orig_freq=orig_freq, new_freq=16_000) # must be a 16 kHz waveform array
+audio_inputs = processor(audios=audio, return_tensors="pt")
+audio_array_from_audio = model.generate(**audio_inputs, tgt_lang="rus")[0].cpu().squeeze()
+```
+
+3. Listen to the audio samples either in an ipynb notebook:
+
+```py
+from IPython.display import Audio
+
+sample_rate = model.sampling_rate
+Audio(audio_array_from_text, rate=sample_rate)
+# Audio(audio_array_from_audio, rate=sample_rate)
```

Comment:
remove this?

Reply:
This was in the initial instructions to cover both use cases (t2st or s2st), I can remove it.

Reply:
Ah no sorry you can keep this!